### PR TITLE
docs: document required intents

### PR DESCRIPTION
**Pull request by advaith1**
**Please describe the changes this PR makes and why it should be merged:**

This PR adds the Gateway Intent required, or the impact of Intents, to descriptions of events and properties.

**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [x] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/client/actions/MessageReactionRemove.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ class MessageReactionRemove extends Action {
     reaction._remove(user);
     /**
      * Emitted whenever a reaction is removed from a cached message.
+     * Requires the GUILD_MESSAGE_REACTIONS or DIRECT_MESSAGE_REACTIONS intent.
```
Comment (vladfrangu) [inline src/client/actions/MessageReactionRemove.js:34]: Might need to specify which is for what (so people don't expect DIRECT_MESSAGE_* to return things from guilds)

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/client/actions/MessageReactionRemove.js (hunk 1 above)
Comment (advaith1) [reply src/client/actions/MessageReactionRemove.js:34]: isn't that kinda common sense though

Comment (vladfrangu) [reply src/client/actions/MessageReactionRemove.js:34]: You'd be surprised... 😓

**Comment by Papaia** [comment]
I don't think this is something the library has to document, it is documented in the Discord gateway reference, which the Intents class and guide return to.
We might want to document which event is which (`GUILD_MEMBER_ADD` <-> `guildMemberAdd`), but that seems pretty clear.

**Comment by advaith1** [comment]
@Papaia I mainly did this to document it on *properties* because many properties returned by discord.js are actually updated by events which aren't obvious (for example, guild.memberCount doesn't get updated without the members intent); since these are created by discord.js they can't be in Discord docs.

And I also documented them on the events because no point in doing it for some but not others

**Comment by advaith1** [comment]
why was this closed? I thought Papaia had accepted my explanation, and it's important to document how intents can affect properties made by the lib

**Comment by iCrawl** [comment]
We never documented anything related to permissions so far, so I don't really see why we would document intents now.

As long as the permissions don't get documented, we prolly won't document intents. Feels 50/50

**Comment by advaith1** [comment]
Would the intents page in the guide be a good place to put this?

**Comment by iCrawl** [comment]
That sounds like a more reasonable approach to me.

**Pull request closed without merging**